pkg/authn/principal.go

```diff
@@ -0,0 +1,87 @@
+// Package authn holds EASi authentication related primitives
+package authn
+
+import "fmt"
+
+const anonID = "ANON"
+
+// ANON is functionally a singleton for representing
+// a request without an identity
+var ANON Principal = (*anonymous)(nil)
+
+// Principal defines the expected behavior for
+// an entity that is making requests of the system.
+type Principal interface {
+	fmt.Stringer
+
+	// ID returns the system identifier
+	// for the given Principal
+	ID() string
+
+	// AllowEASi says whether this principal
+	// is authorized to operate within EASi
+	AllowEASi() bool
```

Comment:
Is this one necessary? If there is an EUAID, then it means a user has authenticated with Okta.

Reply:
Yes, this is a logical extension of having the negative case be an actual object... For the `anonymous` principal, this will be `false`. And the use of the `anonymous` principal will be more clear in later PRs.